app/models/katello/rhel_lifecycle_status.rb

```diff
@@ -0,0 +1,210 @@
+module Katello
+  class RhelLifecycleStatus < HostStatus::Status
+    UNKNOWN = 0
+    FULL_SUPPORT = 1
+    MAINTENANCE_SUPPORT = 2
+    APPROACHING_END_OF_MAINTENANCE = 3
+    EXTENDED_SUPPORT = 4
+    APPROACHING_END_OF_SUPPORT = 5
+    SUPPORT_ENDED = 6
+
+    def self.end_of_day(date)
+      DateTime.parse(date.to_s).end_of_day.utc
+    end
+
+    RHEL_EOS_SCHEDULE = { # dates that each support category ends
+      'RHEL9' => {
+        'full_support' => end_of_day('2027-05-31'),
+        'maintenance_support' => end_of_day('2032-05-31'),
+        'extended_support' => end_of_day('2035-05-31')
+      },
+      'RHEL8' => {
+        'full_support' => end_of_day('2024-05-31'),
+        'maintenance_support' => end_of_day('2029-05-31'),
+        'extended_support' => end_of_day('2032-05-31')
+      },
+      'RHEL7' => {
+        'full_support' => end_of_day('2019-08-06'),
+        'maintenance_support' => end_of_day('2024-06-30'),
+        'extended_support' => end_of_day('2028-06-30')
+      },
+      'RHEL7 (System z (Structure A))' => {
+        'full_support' => end_of_day('2019-08-06'),
+        'maintenance_support' => end_of_day('2021-05-31')
+      },
+      'RHEL7 (ARM)' => {
+        'full_support' => end_of_day('2019-08-06'),
+        'maintenance_support' => end_of_day('2020-11-30')
+      },
+      'RHEL7 (POWER9)' => {
+        'full_support' => end_of_day('2019-08-06'),
+        'maintenance_support' => end_of_day('2021-05-31')
+      },
+      'RHEL6' => {
+        'full_support' => end_of_day('2016-05-10'),
+        'maintenance_support' => end_of_day('2020-11-30'),
+        'extended_support' => end_of_day('2024-06-30')
+      },
+      'RHEL5' => {
+        'full_support' => end_of_day('2013-01-08'),
+        'maintenance_support' => end_of_day('2017-03-31'),
+        'extended_support' => end_of_day('2020-11-30')
+      }
+    }.freeze
+
+    def self.between_dates?(date1, date2)
+      return nil unless date1.present? && date2.present?
+      today = Date.today.beginning_of_day.utc
+      date1.present? && today >= date1.beginning_of_day.utc && today <= date2.end_of_day.utc
+    end
+
+    EOS_WARNING_THRESHOLD = 1.year
+
+    def self.status_map
+      map = {
+        full_support: FULL_SUPPORT,
+        maintenance_support: MAINTENANCE_SUPPORT,
+        approaching_end_of_maintenance: APPROACHING_END_OF_MAINTENANCE,
+        extended_support: EXTENDED_SUPPORT,
+        approaching_end_of_support: APPROACHING_END_OF_SUPPORT,
+        support_ended: SUPPORT_ENDED
+      }
+
+      map.default = UNKNOWN
+      map
+    end
+
+    def self.to_status(operatingsystem: nil)
+      return UNKNOWN unless operatingsystem.is_a?(::Operatingsystem)
+      release = operatingsystem.rhel_eos_schedule_index
```

Comment:
This release schedule index is passed around a lot. 
Is it better to save it into an instance variable? Hopefully the code could be cleaner.

Reply:
If it were an instance method, I'd agree. But this is a class method (`def self.`) and I wouldn't want to store the same eos_schedule_index for the entire class..

Reply:
I do have `def rhel_eos_schedule_index` as an instance method.